File: app/scripts/models/film.js
NziffPicks.Film = DS.Model.extend({
  title: DS.attr(''),
  year: DS.attr(''),
  genres: DS.attr(''),
  runtime: DS.attr(''),
  synopsis: DS.attr(''),
  critics_score: DS.attr(''),
  audience_score: DS.attr(''),
  directors: DS.attr(''),
  image_m: DS.attr(''),
  image_l: DS.attr(''),
  rt_link: DS.attr(''),
  reviews: DS.hasMany('review', { async: true }),
  region: DS.belongsTo('region', { async: true }),

  rank: Ember.computed(function() {
    if (this.get('critics_score') === -1) {
      return this.get('audience_score');
    } else {
      return this.get('critics_score');
    }
  }),

  // goodReview: function() {
  //   // debugger;
  //   return this.get('reviews.length');
  // }.property('reviews.@each')

});

File: importer/lib/film.rb
require_relative 'rottentomatoes'

class Film
  def initialize(params)

    # save properties of payload that are exactly how we want them
    @film = params.slice(:id, :title, :year, :genres, :runtime, :synopsis, :tag, :slug)

    # flatten other properties
    @film[:critics_score] = params[:ratings][:critics_score]
    @film[:audience_score] = params[:ratings][:audience_score]

    @film[:directors] = Array(params[:abridged_directors]).map{|d|d[:name]}

    unless params[:posters][:original].match(/poster_default/)
      @film[:image_m] = params[:posters][:detailed]
      @film[:image_l] = params[:posters][:original]
    end

    @film[:rt_link] = params[:links][:alternate]

    @film[:region] = "auckland"

    # these are the URIs to grab next
    clips_link = params[:links][:clips]

    @film[:reviews] = Review.fetch_and_save_reviews(@film[:id])

    # from links grab :alternate, :clips, :reviews (although clips isn't that great)
  end

  def save
    puts "Saving film"
    # TODO merge with existing imported film
    File.open("imported/films/rotten_tomatoes/#{@film[:id]}.json", 'w') {|f| f.write(ActiveSupport::JSON.encode(@film)) }
  end

end
